### ISSUE TITLE
Documentation sample code for STDP with Gradient Descent possibly incorrect

### ISSUE THREAD

**Issue by and-rewsmith** [fangwei123456/spikingjelly#345]
**Issue type**

- [x] Bug Report
- [ ] Feature Request
- [ ] Help wanted
- [ ] Other

**SpikingJelly version**

`0.0.0.0.13`

**Description**

@AllenYolk Pinging for your awareness.

Documentation and sample code is [provided here](https://spikingjelly.readthedocs.io/zh_CN/latest/activation_based_en/stdp.html#combine-stdp-learning-with-gradient-descent) showing how to combine stdp learning with gradient descent.

I took this sample code, fleshed it out a bit to train on MNIST dataset. However the loss does not go down. In troubleshooting the issue, the loss does in fact go down when I remove the stdp learning and `IFNode`s from the network. This may indicate a conceptual problem with that sample code, or maybe a misunderstanding on my part.

...

**Minimal code to reproduce the error/bug**

Link to constantly reproducing behavior:
https://github.com/and-rewsmith/ml-explorations/blob/main/03-simple-snn-stdp/src/stdp_with_gd_mnist.py#L90

I am happy to answer questions to accelerate any investigations.

**Comment by fangwei123456** [comment]
Hi, STDP is an unspvervised learning method that it does not guarantee that the loss will decrease.
